### PR TITLE
Add volume of hollow circular cylinder, Exceptions

### PR DESCRIPTION
**Pull request by rohanr18**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by CaedenPH — APPROVED** [review]

File: maths/volume.py [hunk 1]
```diff
@@ -11,23 +10,37 @@
...
     >>> vol_cube(1)
     1.0
     >>> vol_cube(3)
     27.0
+    >>> vol_cube(-1)
+    Traceback (most recent call last):
+        ...
+    ValueError: vol_cube() only accepts non-negative values
     """
+    if side_length < 0:
```
Comment (cclauss) [inline maths/volume.py:22]: Please write a test for 0 and for 1.6

File: maths/volume.py [hunk 2]
```diff
@@ -11,23 +10,37 @@
...
+    >>> vol_spherical_cap(-1, 2)
+    Traceback (most recent call last):
+        ...
+    ValueError: vol_spherical_cap() only accepts non-negative values
+    >>> vol_spherical_cap(1, -2)
+    Traceback (most recent call last):
+        ...
+    ValueError: vol_spherical_cap() only accepts non-negative values
     """
+    if height < 0 or radius < 0:
```
Comment (cclauss) [inline maths/volume.py:42]: Please write a test for 0 and for 1.6

File: maths/volume.py (hunk 1 above)
Comment (cclauss) [reply maths/volume.py:22]: Repeat below.

File: maths/volume.py [hunk 3]
```diff
@@ -213,6 +347,9 @@ def main():
     print("Sphere: " + str(vol_sphere(2)))  # ~= 33.5
     print("Hemisphere: " + str(vol_hemisphere(2)))  # ~= 16.75
     print("Circular Cylinder: " + str(vol_circular_cylinder(2, 2)))  # ~= 25.1
+    print(
```
Comment (cclauss) [inline maths/volume.py:350]: Could we do these with f-strings instead?

**Comment by rohanr18** [comment]
@cclauss Anything more?

**Comment by rohanr18** [comment]
@cclauss Is this ready to merge?

**Comment by rohanr18** [comment]
@cclauss .

**Review by cclauss — APPROVED** [review]
Nice improvements!  Thanks for doing this.

**Pull request merged**